events/handler.go

```diff
@@ -47,6 +48,7 @@ type events struct {
 	// satisfy the request of a given watch object in watchers, the event
 	// is sent over the channel to that caller of WatchEvents
 	watchers []*watch
+	eventManagerLock sync.RWMutex
```

Comment:
Add a comment that it protects only eventlist and not watchers, if that is the intent.